### PR TITLE
Fixes | Index

### PR DESCRIPTION
**Pull request by basdag**
Fixed invalid methods in the index file.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -115,24 +108,55 @@ Hook.prototype.parse = function parse() {
   }
 
   //
+  // Start checking linter configuration.
+  //
+  var existLinterConfigurationFile = true;
+
+  try {
+      fs.accessSync(path.resolve(this.root, '.eslintrc.js'));
```
Comment (nicktate) [inline index.js]: `this.root` // The root location of the .git folder.

Are you sure you want to resolve the linter configuration file based on the root `.git` folder?

File: index.js [hunk 2]
```diff
@@ -115,24 +108,55 @@ Hook.prototype.parse = function parse() {
...
+  } catch (e) {
+      existLinterConfigurationFile = false;
+  }
+
+  //
   // Error out if the are linter files already predifined in the root path of the project.
   //
-  if (exists(linterConfigFileProject)) {
+  if (!existLinterConfigurationFile) {
```
Comment (nicktate) [inline index.js]: This logic isn't checking what you intended. You still want to be looking for existence of project configuration not the linter configuration.

File: index.js (hunk 1 above)
Comment (basdag) [inline index.js]: its a logical place to be the root, since one level up from this is outside what you can push

File: index.js (hunk 2 above)
Comment (basdag) [inline index.js]: It doing was I plan, its looking to see if a eslinter configuration file is on the root folder of the project.

Comment (nicktate) [inline index.js]: You say if it does not exist then error? When that should be the valid case.

**Pull request closed without merging**